pkg/controllers/cloud_config_sync_controller_test.go

```diff
@@ -164,24 +174,12 @@ var _ = Describe("Cloud config sync controller", func() {
 			GracePeriodSeconds: pointer.Int64(0),
 		}
 
-		if infraCloudConfig != nil {
-			Expect(cl.Delete(ctx, infraCloudConfig, deleteOptions)).To(Succeed())
-			Eventually(
-				apierrors.IsNotFound(cl.Get(ctx, client.ObjectKeyFromObject(infraCloudConfig), &corev1.ConfigMap{})),
-			).Should(BeTrue())
-		}
-
-		if managedCloudConfig != nil {
-			Expect(cl.Delete(ctx, managedCloudConfig, deleteOptions)).To(Succeed())
+		allCMs := &corev1.ConfigMapList{}
+		Expect(cl.List(ctx, allCMs)).To(Succeed())
+		for _, cm := range allCMs.Items {
+			Expect(cl.Delete(ctx, &cm, deleteOptions)).To(Succeed())
 			Eventually(
-				apierrors.IsNotFound(cl.Get(ctx, client.ObjectKeyFromObject(managedCloudConfig), &corev1.ConfigMap{})),
-			).Should(BeTrue())
-		}
-
-		if syncedCloudConfigMap != nil {
-			Expect(cl.Delete(ctx, syncedCloudConfigMap, deleteOptions)).To(Succeed())
-			Eventually(
-				apierrors.IsNotFound(cl.Get(ctx, client.ObjectKeyFromObject(syncedCloudConfigMap), &corev1.Namespace{})),
+				apierrors.IsNotFound(cl.Get(ctx, client.ObjectKeyFromObject(&cm), &corev1.ConfigMap{})),
```

Comment:
As mentioned in slack, but for the benefit or future selves. You could also use a `DeleteAllOf` call here, which will delete all of the configmaps in a particular namespace. For example this could be done:
```
By("Cleanup resources")
for _, ns := range []string{targetNamespaceName, OpenshiftConfigNamespace, OpenshiftManagedConfigNamespace} {
	Expect(cl.DeleteAllOf(ctx, &corev1.ConfigMap{},
		client.GracePeriodSeconds(0),
		client.InNamespace(ns),
	)).To(Succeed())
	Eventually(func() (bool, error) {
		cmList := &corev1.ConfigMapList{}
		if err := cl.List(ctx, cmList, client.InNamespace(ns)); err != nil {
			return false, err
		}
		return len(cmList.Items) == 0, nil
	},
	).Should(BeTrue())
}
```